pkg/repositories/gormimpl/list.go

```diff
@@ -28,14 +37,21 @@ func applyListModelsInput(tx *gorm.DB, sourceEntity common.Entity, in models.Lis
 		return nil, errors.GetInvalidEntityError(sourceEntity)
 	}
 
-	sourceTableName := tx.NewScope(sourceModel).TableName()
+	sourceTableName, err := getTableName(tx, sourceModel)
+	if err != nil {
+		return nil, err
```

Comment:
can we wrap this in a DataCatalogError so we can add a grpc reponse code too? https://github.com/flyteorg/datacatalog/blob/master/pkg/errors/errors.go#L44:6

Reply:
Fixed.